### PR TITLE
Nerfs insulated gloves (by nerfing shocks)

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

The 8 second paralysis from being electrocuted has been replaced by an 8 second knockdown. The length of this knockdown can be reduced by having a low siemens coefficient (and it isn't applied at all if you're wearing insulated gloves or the like).

The Spell Packets wizard spell still hardstuns you if you're electrocuted by it, and it now has the SHOCK_NOGLOVES flag, like other shocks that don't enter your body through your arms.

Revenant light overloads also still hardstun, because it's hard for them to capitalize upon it and it's an important part of their kit.

The conditions for what does and doesn't cause a shock knockdown have been made slightly more clear in the code.

 I feel like I shouldn't have to explicitly say this, but if I know that if I don't, the comments will descend into a flame war: To be 100% clear, this PR does not actually make any direct changes to insulated gloves, and they still provide immunity to the damage and knockdowns of most shocks. The title of this PR was a joke to draw attention.

## Why It's Good For The Game

Being rendered helpless for 8 seconds because you dared to not rush for a form of shock immunity feels awful. You can still be dunked on for touching a shocked door (shovestun go brrr), but the kill (or handcuffing) isn't _as_ free as it was before.

The title was mostly a joke, but this could genuinely reduce the usage rate of insulated gloves, as the punishment for not getting them is now much less harsh.

Shocks knocking down instead of hardstunning could also allow me to re-merge liquid electricity and enriched liquid electricity in a future PR, as the latter would be much weaker.

Spell Packets still hardstuns because the hardstun from the illusory electrocution was, like, the entire point of the spell. I've given it the SHOCK_NOGLOVES flag because the electrocution comes from being hit in the chest by an enchanted packet of birdseed, not touching something with your hand.

This PR has been tested on a local test server.

## Changelog

:cl: ATHATH
balance: We've heard your complains about baton combat, and we have decided to address them by nerfing insulated gloves. The hardstun from being electrocuted is now a knockdown. This should reduce the demand for insulated gloves.
balance: The Spell Packets wizard spell still tazes you, though. It also ignores your gloves now, as the electrocution comes from being hit in the chest by an enchanted packet of birdseed, not touching something with your hand.
balance: Revenant light overloads also still hardstun, because it's hard for them to capitalize upon it and it's an important part of their kit.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
also the burning yellow extract is now an AoE knockdown instead of an AoE hardstun

~~that means this counts as a xenobio nerf, right?~~

**Comment by Mothblocks** [comment]
power creep

**Comment by newfren** [comment]
I would actually love it if I didn't feel like I needed insuls every single round

**Review by MrMelbert — COMMENTED** [review]
don't forget the shock hallucination :)

**Comment by Wallemations** [comment]
Doesn't this mean you'd be able to instantly move and end up shocking yourself multiple times in a row if it's something like an electrified grille? I'm thinking at least a second of stun just to stop the player's momentum entirely so they have a moment to realize they've been shocked would at least be beneficial, but that could entirely be just me.

**Comment by ATH1909** [comment]
> don't forget the shock hallucination :)

does it not call electrocute_act() with a flag to designate it as being an illusion?

**Comment by ATH1909** [comment]
> Doesn't this mean you'd be able to instantly move and end up shocking yourself multiple times in a row if it's something like an electrified grille? I'm thinking at least a second of stun just to stop the player's momentum entirely so they have a moment to realize they've been shocked would at least be beneficial, but I could more just be suggesting that because I'd imagine myself being stupid enough with a slow reaction time to do that.

I tried it out on a test server, vending machines and doors didn't seem to instafry me

**Comment by Wallemations** [comment]
> I tried it out on a test server, vending machines and doors didn't seem to instafry me

What about something that doesn't need to be hacked beforehand like electrified grilles?

**Comment by SkeletalElite** [comment]
> What about something that doesn't need to be hacked beforehand like electrified grilles?

I believe shocking has a cooldown to prevent conveyors from spamming shocks too rapidly.

**Comment by Ryusenshu** [comment]
poor malfies

**Comment by githubuser4141** [comment]
> 

What if you're lagging? I can imagine myself easily walking, then crawling into a shocked door until I'm in crit if there's enough power flowing through. I think doors normally have a shock delay, that's less then the hardstun time which'd usually prevent this.

File: code/modules/antagonists/revenant/revenant_abilities.dm [hunk 1]
```diff
@@ -258,8 +258,8 @@
 		if(human_mob == caster)
 			continue
 		to_shock.Beam(human_mob, icon_state = "purple_lightning", time = 0.5 SECONDS)
-		if(!human_mob.can_block_magic(antimagic_flags))
-			human_mob.electrocute_act(shock_damage, to_shock, flags = SHOCK_NOGLOVES)
+		if(!human_mob.can_block_magic(antimagic_flags) && human_mob.electrocute_act(shock_damage, to_shock, flags = SHOCK_NOGLOVES))
+			human_mob.Paralyze(80)
```
Comment (Fikou) [inline code/modules/antagonists/revenant/revenant_abilities.dm:262]: reaction on athaths face when i tell him about seconds defines

**Comment by XDTM** [comment]
> > don't forget the shock hallucination :)
> 
> does it not call electrocute_act() with a flag to designate it as being an illusion?
> 
> oh my god it doesn't what the fuck

It needs to be invisible to onlookers, not just do fake damage

File: code/modules/antagonists/revenant/revenant_abilities.dm (hunk 1 above)
Comment (ATH1909) [reply code/modules/antagonists/revenant/revenant_abilities.dm:262]: :walter:

**Comment by JohnFulpWillard** [comment]
I disagree with this PR because stuns from being shocked is both important for item balance and because it's the pinnacle of slapstick comedy.

It also limits the AI from helping catch people alive (both for sec and antags) if the AI isn't Asimov, which is something we should consider because as a sandbox we shouldn't expect the AI to be on the same lawset every round.

**Comment by ATH1909** [comment]
resolved melbert and fikou's requests

**Comment by ATH1909** [comment]
> I disagree with this PR because stuns from being shocked is both important for item balance and because it's the pinnacle of slapstick comedy.
> 
> It also limits the AI from helping catch people alive (both for sec and antags) if the AI isn't Asimov, which is something we should consider because as a sandbox we shouldn't expect the AI to be on the same lawset every round.

I'd argue that shocks not being instant ggs in fights probably gives more room for (and is healthier for) item balance. See: the xenobio crossbreed that electrocutes everyone within 4 tiles of you, which would otherwise need to be balanced as "stuns everyone within 4 tiles of you" or "mildly damages everyone within 4 tiles of you", with little granularity in-between. Abilities and items that were balanced around hardstunning people they electrocuted can just be given a hardstun upon electrocution success directly, as I've demonstrated with Spell Packets and revenant lightning.

As for the latter point, that's kind of why I made this PR- shocking from an enemy AI who's not on asimov is currently incredibly oppressive if you haven't gone out of your way to grab a way to no-sell it. As a knockdown, it's still a powerful tool, especially if you're working with humans, just not one that allows a borg to casually walk up to a shocked antag, construct a locker around them, and weld it shut before they recover from the stun.

**Comment by Pickle-Coding** [comment]
This is pointless.

First, a nice shocking animation is shown, showing you your skeleton and making you jiggle. This is a good "oh fuck" moment while the shock animation is shown. After the shock animation is done, you are still on the floor, jiggling as your muscles are seized. This gives you time to think about your actions, and displays a good "you fucked up" moment for you to experience.

Turning it into a knockdown means people have the option to move. This nullifies the message the zap gave to you, and gives people no time to think about their actions as they can immediately move away.

Getting shocked from a grille is almost always the players fault, either for walking into it or letting themselves get shoved in such a dangerous area. From an airlock, same story, with an extra fuck you to tiders. Malfunctioning AIs are supposed to be scary, so paralysis helps increase the fear as the player loses control of themselves.

Apparently some antagonist causes of zaps were made to still give a paralysis, so I honestly have no idea what's the point in this other than to make getting shocked a less interesting experience.

**Comment by Redrover1760** [comment]
Ruins numerous deathtraps involving electrified grilles and other fun and interesting concepts by removing the stun altogether, as well as a nerf to malf ai and just fun shocking things in general, which is completely unnecessary, no one has complained about it. Do not merge this.

**Comment by optimumtact** [comment]
swing and a miss

**Comment by Redrover1760** [comment]
> > I disagree with this PR because stuns from being shocked is both important for item balance and because it's the pinnacle of slapstick comedy.
> > It also limits the AI from helping catch people alive (both for sec and antags) if the AI isn't Asimov, which is something we should consider because as a sandbox we shouldn't expect the AI to be on the same lawset every round.
> 
> I'd argue that shocks not being instant ggs in fights probably gives more room for (and is healthier for) item balance. See: the xenobio crossbreed that electrocutes everyone within 4 tiles of you, which would otherwise need to be balanced as "stuns everyone within 4 tiles of you" or "mildly damages everyone within 4 tiles of you", with little granularity in-between. Abilities and items that were balanced around hardstunning people they electrocuted can just be given a hardstun upon electrocution success directly, as I've demonstrated with Spell Packets and revenant lightning.
> 
> As for the latter point, that's kind of why I made this PR- shocking from an enemy AI who's not on Asimov is currently incredibly oppressive if you haven't gone out of your way to grab a way to no-sell it. As a knockdown, it's still a powerful tool, especially if you're working with humans, just not one that allows a borg to casually walk up to a shocked antag, construct a locker around them, and weld it shut before they recover from the stun.

Have you considered instead giving the RD a way to locate active AIs, then?

**Pull request closed without merging**